src/dbnode/storage/index/mutable_segments.go

```diff
@@ -239,6 +240,34 @@ func (m *mutableSegments) seriesActive(d doc.Metadata) bool {
 	return !d.OnIndexSeries.TryMarkIndexGarbageCollected()
 }
 
+func (m *mutableSegments) OnNotContainsDoc(d doc.Metadata) {
+	// Track expired series filtered out from new index segment during compaction.
+	m.metrics.activeBlockGarbageCollectSeries.Inc(1)
```

Comment:
Especially since this is just for metrics, can we just remove `OnNotContainsDoc` and do the increment in `ContainsDocs`?